res/controllers/Reloop-Jockey-3-ME-scripts.js

```diff
@@ -0,0 +1,444 @@
+}
+
+Jockey3ME.loop_double_halve = function (channel, control, value, status, group) {
+  var newValue = (value-64);
+
+  if (newValue == 1) {
+    engine.setValue(group,"loop_double",1);
+    engine.setValue(group,"loop_double",0);
+  } else {
+    engine.setValue(group,"loop_halve",1);
+    engine.setValue(group,"loop_halve",0);
+  }
+}
+
+Jockey3ME.move_beat = function (channel, control, value, status, group) {
+  var newValue = (value-64);
+  engine.setValue(group,"beatjump",(newValue*Jockey3ME.move_beat_value));
+}
+
+Jockey3ME.crossfader = function (channel, control, value, status, group) {
+  var newValue = 0;
+  if (control == 0x37 && !Jockey3ME.crossfaderScratch) {
+    newValue = ((value / 63.5) - 1);
+    engine.setValue(group,"crossfader",newValue);
+  } else if (control == 0x37 && Jockey3ME.crossfaderScratch) {
+    switch (value) {
+      case 127:
+        engine.setValue(group,"crossfader",1);
+        break;
+      case 126:
+        engine.setValue(group,"crossfader",0.96875);
+        break;
+      case 125:
+        engine.setValue(group,"crossfader",0.953125);
+        break;
+      case 2:
+        engine.setValue(group,"crossfader",-0.953125);
+        break;
+      case 1:
+        engine.setValue(group,"crossfader",-0.96875);
+        break;
+      case 0:
+        engine.setValue(group,"crossfader",-1);
+        break;
+      default:
+        engine.setValue(group,"crossfader",0);
+    }
+  } else {
+    if (value <= 126) {
+      script.crossfaderCurve(value, 0, 126);
+      Jockey3ME.crossfaderScratch = false;
+    } else {
+      Jockey3ME.crossfaderScratch = true;
+    }
+  }
+}
+
+Jockey3ME.CUP = function (channel, control, value, status, group) {
+  if (value == 0x7F) {
+    engine.setValue(group,"cue_default",1);
+    engine.setValue(group,"cue_default",0);
+    midi.sendShortMsg(status,control,127);
+  } else {
+    engine.setValue(group,"play",1);
+    midi.sendShortMsg(status,control,0);
+  }
+}
+
+Jockey3ME.MixerVol = function (channel, control, value, status, group) {
+  var currentDeck = parseInt(group.substring(8,9));
+  if ((Jockey3ME.MixerDeck1 == 1 && currentDeck == 1) || (Jockey3ME.MixerDeck2 == 1 && currentDeck == 2)) {
+    currentDeck += 2;
+  }
+  if (control == 0x2D || control == 0x6C) {
+	var noGainHop = engine.getParameter("[Channel" + currentDeck + "]","pregain");
+	if (!(!((noGainHop - (value / 127)) < 0.04) || !((noGainHop - (value / 127)) > -0.04))) {
+		Jockey3ME.noVolHopValue[1] = false;
+	}
+	if (!Jockey3ME.noVolHopValue[1]) {
+		engine.setParameter("[Channel" + currentDeck + "]","pregain",(value / 127));
+    }
+  } else {
+    var noVolHop = engine.getParameter("[Channel" + currentDeck + "]","volume");
+    if (!(!((noVolHop - (value / 127)) < 0.04) || !((noVolHop - (value / 127)) > -0.04))) {
+      Jockey3ME.noVolHopValue[0] = false;
+    }
+    if (!Jockey3ME.noVolHopValue[0]) {
+      engine.setParameter("[Channel" + currentDeck + "]","volume",(value / 127));
+    }
+  }
+}
```

Comment:
Why does this require JS? Couldn't this be done with a simple XML mapping with the soft-takeover option?